sopel/modules/find.py

```diff
@@ -113,38 +114,58 @@ def kick_cleanup(bot, trigger):
 
 # Match nick, s/find/replace/flags. Flags and nick are optional, nick can be
 # followed by comma or colon, anything after the first space after the third
-# slash is ignored, you can escape slashes with backslashes, and if you want to
-# search for an actual backslash followed by an actual slash, you're shit out of
-# luck because this is the fucking regex of death as it is.
+# slash is ignored, and you can use either a slash or a pipe.
+# If you want to search for an actual slash AND a pipe in the same message,
+# you can escape your separator, in old and/or new.
 @plugin.rule(r"""(?:
-            (\S+)           # Catch a nick in group 1
-          [:,]\s+)?         # Followed by colon/comma and whitespace, if given
-          s/                # The literal s/
-          (                 # Group 2 is the thing to find
-            (?:\\/ | [^/])+ # One or more non-slashes or escaped slashes
-          )/(               # Group 3 is what to replace with
-            (?:\\/ | [^/])* # One or more non-slashes or escaped slashes
-          )
-          (?:/(\S+))?       # Optional slash, followed by group 4 (flags)
-          """)
+             (?P<nick>\S+)    # Catch a nick in group 1
+             [:,]\s+)?        # Followed by optional colon/comma and whitespace
+             s(?P<sep>/)      # The literal s and a separator / as group 2
+             (?P<old>         # Group 3 is the thing to find
+               (?:\\/|[^/])+  # One or more non-slashes or escaped slashes
+             )
+             /                # The separator again
+             (?P<new>         # Group 4 is what to replace with
+               (?:\\/|[^/])*  # One or more non-slashes or escaped slashes
+             )
+             (?:/             # Optional separator followed by group 5 (flags)
+                (?P<flags>\S+)
+             )?
+            """)
+@plugin.rule(r"""(?:
+             (?P<nick>\S+)    # Catch a nick in group 1
+             [:,]\s+)?        # Followed by optional colon/comma and whitespace
+             s(?P<sep>\|)     # The literal s and a separator | as group 2
+             (?P<old>         # Group 3 is the thing to find
+               (?:\\\||[^|])+  # One or more non-slashes or escaped slashes
```

Comment:
It's not a slash, though…?

```suggestion
               (?:\\\||[^|])+  # One or more non-pipe or escaped pipe
```